pep/app/authenticators/ip.ts

```diff
@@ -0,0 +1,53 @@
+import { inject as service } from '@ember/service';
+import { isEmpty } from '@ember/utils';
+import { resolve, reject } from 'rsvp';
+import BaseAuthenticator from 'ember-simple-auth/authenticators/base';
+
+import AjaxService from 'pep/services/ajax';
+import ENV from 'pep/config/environment';
+import { PepSecureAuthenticatedData } from 'pep/api';
+export default class IpAuthenticator extends BaseAuthenticator {
+    @service ajax!: AjaxService;
+    authenticationHeaders = {
+        'Content-Type': 'application/json'
+    };
+
+    /**
+     * Authenticates and logs the user in
+     *
+     * @param {string} sessionId
+     * @returns
+     * @memberof IpAuthenticator
+     */
+    async authenticate(sessionId: string) {
+        const result = await this.ajax.request(`${ENV.authBaseUrl}/Authenticate/ip?SessionId=${sessionId}`, {
+            headers: this.authenticationHeaders
+        });
+        return result;
+    }
+
+    /**
+     * Invalidates the local session and logs the user out
+     */
+    invalidate(data: PepSecureAuthenticatedData) {
+        return this.ajax.request(`${ENV.authBaseUrl}/Users/Logout`, {
+            method: 'POST',
+            headers: this.authenticationHeaders,
+            body: this.ajax.stringifyData({
+                SessionId: data.SessionId
+            })
+        });
+    }
+
+    /**
+     * Restores the local session from cookies, if one exists
+     * @param {SessionAuthenticatedData} data
+     */
+    restore(data: PepSecureAuthenticatedData) {
+        if (!isEmpty(data.SessionId)) {
+            return resolve(data);
+        } else {
+            return reject();
+        }
+    }
+}
```

Comment:
@bakerac4 we probably should just keep the IP-based auth in the existing `pep` authenticator, as its more or less identical besides the `authenticate()` method, which we can customize by adding an optional `grantType: 'password' | 'ip'` arg. 

I think this is preferable, as its consistent with our patterns in other projects and how the built-in oAuth authenticator handles different "grant types", which is what this really is, just another way to auth to the same provider. But if you really want to make it a separate authenticator, we should at least extend from the `pep` authenticator, so we aren't duplicating the other logic (invalidate, restore, and also the session expiration timer logic once you add it).

Also the authenticate() method probably shouldn't have an arg for `sessionId`, since per our discussion yesterday, if we are storing that "unauthed session id" in a separate cookie, the authenticator could inject the cookies service and read it (and set it if ip-based auth "fails") as needed to keep that all self-contained.